rules/TypeDeclaration/TypeInferer/PropertyTypeInferer/TrustedClassMethodPropertyTypeInferer.php

```diff
@@ -73,7 +73,7 @@ public function inferProperty(Class_ $class, Property $property, ClassMethod $cl
 
         $resolvedTypes = [];
         foreach ($assignedExprs as $assignedExpr) {
-            $resolvedTypes[] = $this->nodeTypeResolver->getType($assignedExpr);
+            $resolvedTypes[] = $this->nodeTypeResolver->getNativeType($assignedExpr);
```

Comment:
thats a actual bugfix for https://github.com/rectorphp/rector-src/pull/5018#issuecomment-1719578352

Reply:
Could you cherry pick this patch in separate PR? Thank you.